src/google/protobuf/map_entry.h

```diff
@@ -248,8 +248,8 @@ class MapEntry : public MapEntryBase {
         entry,
         offsets_,
         has_bits_,
-        GOOGLE_PROTOBUF_GENERATED_MESSAGE_FIELD_OFFSET(MapEntry, entry_lite_._has_bits_),
-        GOOGLE_PROTOBUF_GENERATED_MESSAGE_FIELD_OFFSET(MapEntry, _internal_metadata_),
+        static_cast<int>(GOOGLE_PROTOBUF_GENERATED_MESSAGE_FIELD_OFFSET(MapEntry, entry_lite_._has_bits_)),
```

Comment:
I have made an attempt to change the type of these two ReflectionSchema fields from int to uint32, but then many more changes have become required. My proposal is to merge this CL to fix the breakage under old Clang and therefore unblock Asad, and do this cleanup separately.